### ISSUE TITLE
Empty Array Literals cause ArrayIndexOutofBounds

### ISSUE THREAD

**Issue by daumayr** [smarr/SOMns#216]
The use of Empty Array literals (`{}`) to create Arrays of size 0 is causing ArrayIndexOutOfBoundsExceptions, for example in:

```
public main: args = (
  {}.
  ^ 1
)
```

Stack:

```
java.lang.ArrayIndexOutOfBoundsException: 0
	at som.primitives.arrays.ArraySetAllStrategy.evaluateFirstDetermineStorageAndEvaluateRest(ArraySetAllStrategy.java:258)
	at som.interpreter.nodes.literals.ArrayLiteralNode$Uninit.executeGeneric(ArrayLiteralNode.java:43)
	at som.interpreter.nodes.SequenceNode.executeAllButLast(SequenceNode.java:50)
	at som.interpreter.nodes.SequenceNode.executeGeneric(SequenceNode.java:43)
	at som.interpreter.Invokable.execute(Invokable.java:51)
	at org.graalvm.compiler.truffle.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:250)
	at org.graalvm.compiler.truffle.OptimizedCallTarget.callRoot(OptimizedCallTarget.java:239)
	at org.graalvm.compiler.truffle.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:229)
	at org.graalvm.compiler.truffle.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:214)
	at org.graalvm.compiler.truffle.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:198)
	at org.graalvm.compiler.truffle.OptimizedDirectCallNode.callProxy(OptimizedDirectCallNode.java:65)
	at org.graalvm.compiler.truffle.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:59)
	at som.interpreter.nodes.dispatch.CachedDispatchNode.executeDispatch(CachedDispatchNode.java:38)
```

**Comment by smarr** [comment]
@richard-roberts something we overlooked.
I think, we can keep this one around to get students started. @daumayr workaround for the moment is `Array new: 0`.

**Comment by smarr** [comment]
I got annoyed with this one, so, I just fixed it.

**Comment by smarr** [comment]
This is solved with #215.
